### PR TITLE
DAOS-9873 test: Fix more pool tests

### PR DESCRIPTION
**Pull request by mjmac**
The following tests were using the check_for_pool() helper
which did not use sudo when looking into SCM:
  - pool/multi_server_create_delete.py
  - pool/dynamic_server_pool.py

Fix the check_for_pool() helper to use the sudo-aware
check_file_exists() helper, and to return a friendlier
bool value instead of a return code.

Features: pool control

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

**Review by daltonbohning — CHANGES_REQUESTED** [review]
The commit pragma `Feature: pool control` should be plural: `Features: pool control`

File: src/tests/ftest/pool/dynamic_server_pool.py [hunk 1]
```diff
@@ -62,8 +62,7 @@ def check_pool_location(self, hosts, uuid_to_ranks):
             # appears that self.hostlist_servers[0] is always rank0, 1 is rank1,
             # and the extra server we'll be adding will be rank2.
             for rank, host in enumerate(hosts):
-                rc = check_for_pool(host, pool.uuid.lower())
-                pool_exists_on_host = rc == RC_SUCCESS
+                pool_exists_on_host = check_for_pool(host, pool.uuid.lower())
```
Comment (daltonbohning) [inline src/tests/ftest/pool/dynamic_server_pool.py:65]: We should also remove the instantiation of `RC_SUCCESS` as it's no longer used:
https://github.com/daos-stack/daos/blob/c1d11e9d74a059249eec29cc1be808c1e7ee6baa/src/tests/ftest/pool/dynamic_server_pool.py#L58

**Comment by mjmac** [comment]
> The commit pragma `Feature: pool control` should be plural: `Features: pool control`

D'oh! I've only been using the pragma for months now... Thanks, fixed.

File: src/tests/ftest/pool/dynamic_server_pool.py [hunk 2]
```diff
@@ -100,7 +99,7 @@ def test_dynamic_server_pool(self):
 
         :avocado: tags=all,full_regression
         :avocado: tags=vm
-        :avocado: tags=control,dynamic_server_pool
+        :avocado: tags=pool,control,dynamic_server_pool
```
Comment (daltonbohning) [inline src/tests/ftest/pool/dynamic_server_pool.py:102]: This is okay, but we prefer to put "features" on a separate line
```suggestion
        :avocado: tags=pool,control
        :avocado: tags=dynamic_server_pool
```

**Review by daltonbohning — APPROVED** [review]
Minor suggestions if you need to push again

File: src/tests/ftest/pool/dynamic_server_pool.py (hunk 2 above)
Comment (mjmac) [reply src/tests/ftest/pool/dynamic_server_pool.py]: Done, was wondering about that.

File: src/tests/ftest/pool/dynamic_server_pool.py (hunk 1 above)
Comment (mjmac) [reply src/tests/ftest/pool/dynamic_server_pool.py:64]: Removed.

File: src/tests/ftest/pool/dynamic_server_pool.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/pool/dynamic_server_pool.py]: Thanks!

**Review by daltonbohning — APPROVED** [review]

**Review by shimizukko — DISMISSED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**